src/include/duckdb/storage/checkpoint/table_data_writer.hpp

```diff
@@ -43,7 +40,7 @@ class TableDataWriter {
 
 protected:
 	DuckTableEntry &table;
-	optional_ptr<ClientContext> client_context;
+	QueryContext &context;
```

Comment:
This should go back to being the `ClientContext` - we can implicitly convert it to `QueryContext` when required.